### DISCUSSION TITLE
Adding Cert and Key for Unleash HTTPS port

### DISCUSSION THREAD

**Discussion by deepak-grover9** [Unleash/unleash#5687] [Q&A]
I want to run unleash on HTTPS port (8443) ...The cert and key are stored in vault which I am getting at runtime as pem files. Is there a way, can I configure the cert and key for HTTPS port

**Comment by deepak-grover9** [comment]
I have my server file something like below.
const unleash = require('unleash-server');

unleash
  .start({
    db: {
      ssl: false,
      host: 'localhost',
      port: 5432,
      database: 'unleash',
      user: 'unleash_user',
      password: 'password',
    },
    server: {
      port: 8443,
    },
  })
  .then((unleash) => {
    console.log(
      `Unleash started on http://localhost:${unleash.app.get('port')}`,
    );
  });

I am getting secrets as /vault/secrets/cert.pem and /vault/secrets/key.pem ...I want to configure these secrets for 8443 port which is HTTPS...Is there a way I can configure my secrets

**Comment by andreas-unleash** [comment]
Hi @deepak-grover9 , I would suggest you handle your certificates and key by using a reverse proxy (like Nginx).  Is there a particular reason you would like to handle it directly in the express server? (Generally speaking doing it in Nginx is more scalable, secure and will give you a centralised space for your certs)
